### DISCUSSION TITLE
Move Hub: Run motors with button click.

### DISCUSSION THREAD

**Discussion by oneauraaa** [pybricks/support#1257] [Q&A]
Version: `('movehub', '3.3.0b8', 'v1.20.0-23-g6c633a8dd on 2023-07-07')`
Device: LEGO Boost Move Hub

Hello, I already described that but anyways. I have move hub and I want to run the **A, B, C, D** ports with the built-in hub button. **On the first click A then B then C then D then both A and B then both C and D.**

Here is **my working but not doing what I want** code:
```
from pybricks.tools import wait

port = {1: Port.A, 2: Port.B, 3: Port.C, 4: Port.D} #The port
ports = len(port)

value = 1 #The value

theloop = 0 #The loop value

motor = Motor(port[value]) # Initialize a motor on port C.

while (theloop == 0):
    motor.run(500) # Make the motor run clockwise at 500 degrees per second.
```

And in this code **I started to do what I want but much of tries I stuck half of the code so I cant do what I want**:
```
from pybricks.pupdevices import Motor
from pybricks.parameters import Port
from pybricks.tools import wait

port = {1: Port.A, 2: Port.B, 3: Port.C, 4: Port.D} #The port
ports = len(port)

value = 1 #The value

theloop = 0 #The loop value

motor = Motor(port[value]) # Initialize a motor on port C.

while (theloop == 0):
    motor.run(500) # Make the motor run clockwise at 500 degrees per second.
```

My English is not good so my sentences may be inverse.
Thanks.

**Comment by BertLindeman** [comment]
I see that you learn fast.

So let me try to help you by pointing you to parts of the documentation.

By doing that you see more examples that you can pick parts of.
OK?

You had you program snippets already well indicated between ` ``` `  and ` ``` `   
The source is nicely formatted if you surround your program with 
` ```python ` and  ` ``` `  

Did you show the whole program above here?
As I see no code to have the hub button NOT end the program.

Hint 1:  Look at [button-and-system-examples for the move hub](https://beta.pybricks.com/static/docs/v2.9.0/hubs/movehub.html?highlight=set_stop_button#button-and-system-examples)
And add the code to do that to your program.

Hint 2: There is no loop that manipulates the `value` in your program.
Some where `value` needs to have added **one** and if too high, to be set back
Look at your previous program that manipulated the colors.

So far for now.
Let us know your progress.

Bert

**Reply by oneauraaa** [reply]
No, this is all of the code. I stopped coding at the error. I forget to say I have little bit experince at Python.

**Reply by BertLindeman** [reply]
That is why I like to point you to the examples.

Look at them, run them and learn.

Bert
